### PR TITLE
Extensibility

### PR DESCRIPTION
**Pull request by malsbat**
The main part to look at is the changes to Message-Protocol.md.

### CODE REVIEW

File: src/netmcast.c [hunk 1]
```diff
@@ -143,23 +142,26 @@ static DPS_Status MulticastRxInit(DPS_MulticastReceiver* receiver)
...
+        char name[INET6_ADDRSTRLEN + 1];
         /*
          * Filter out interfaces we are not interested in
          */
         if (!UseInterface(receiver->ipVersions, ifn)) {
             continue;
         }
         if (ifn->address.address4.sin_family == AF_INET6) {
-            ret = uv_ip6_name((struct sockaddr_in6*)&ifn->address, addr, sizeof(addr));
+            char ifaddr[INET6_ADDRSTRLEN + + MAX_INTERFACE_NAME_LEN + 2];
```
Comment (GregBurns) [inline src/netmcast.c]: The "+ +" looks like a typo

File: doc/Message-Protocol.md [hunk 2]
```diff
@@ -1,111 +1,129 @@
...
+
+## Subscription message
+
+~~~
+sub = 2
+~~~
+
+*port* and *seq-num* are mandatory in the *unprotected* section.
+
+In a regular subscription message, *sub-flags*, *mesh-id*, *needs*
```
Comment (GregBurns) [inline doc/Message-Protocol.md]: This doesn't look right to me. There is no keying material for sub/sub-acks so the protected section should be empty unless you are thinking of adding support for a mesh-global encryption key for authenticating sub/sub-ack messages. Previous thinking was that sub/sub-acks would be adequately protected by the node-to-node TLS links.

File: src/netmcast.c (hunk 1 above)
Comment (malsbat) [reply src/netmcast.c]: Will fix

File: doc/Message-Protocol.md (hunk 2 above)
Comment (malsbat) [reply doc/Message-Protocol.md]: Agreed, I moved them to the unprotected section in commit 35e1f780cad315ae43f7ad5b8254172e17a7adbb

**Pull request merged**